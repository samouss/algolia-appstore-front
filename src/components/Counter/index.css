.Counter {
  color: #b9b9b9;
  font-size: 0.85rem;
}
