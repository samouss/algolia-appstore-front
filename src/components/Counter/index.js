import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

const Counter = ({ value, customClassName }) => (
  <span
    className={customClassName}
    styleName="Counter"
  >
    {value.toLocaleString()}
  </span>
);

Counter.propTypes = {
  value: PropTypes.number.isRequired,
  customClassName: PropTypes.string,
};

Counter.defaultProps = {
  customClassName: '',
};

export default Counter;
