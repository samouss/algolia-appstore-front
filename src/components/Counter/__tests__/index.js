import React from 'react';
import { shallow } from 'enzyme';
import Counter from '../index';

describe('<Counter />', () => {
  const defaultProps = {
    value: 1235,
  };

  it('expect to render', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <Counter
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render with customClassName', () => {
    const props = {
      ...defaultProps,
      customClassName: 'custom-class-name',
    };

    const component = shallow(
      <Counter
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
