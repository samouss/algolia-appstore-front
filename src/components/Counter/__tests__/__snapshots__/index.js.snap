// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> expect to render 1`] = `
<span
  className=""
  styleName="Counter"
>
  1,235
</span>
`;

exports[`<Counter /> expect to render with customClassName 1`] = `
<span
  className="custom-class-name"
  styleName="Counter"
>
  1,235
</span>
`;
